src/Provider/OpenStreetMap.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/*!
+* HybridAuth
+* https://hybridauth.github.io | https://github.com/hybridauth/hybridauth
+*  (c) 2017 Hybridauth authors | https://hybridauth.github.io/license.html
+*/
+
+namespace Hybridauth\Provider;
+
+use Hybridauth\Adapter\OAuth1;
+use Hybridauth\Exception\UnexpectedApiResponseException;
+use Hybridauth\Data;
+use Hybridauth\User;
+
+/**
+ * OpenStreetMap OAuth1 provider adapter.
+ */
+class OpenStreetMap extends OAuth1
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected $apiBaseUrl = 'https://api.openstreetmap.org/api/0.6/';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $authorizeUrl = 'https://www.openstreetmap.org/oauth/authorize';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $requestTokenUrl = 'https://www.openstreetmap.org/oauth/request_token';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $accessTokenUrl = 'https://www.openstreetmap.org/oauth/access_token';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $apiDocumentation = 'https://wiki.openstreetmap.org/wiki/OAuth';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUserProfile()
+    {
+        $response = $this->apiRequest('user/details');
+
+        $response = new \SimpleXMLElement($response);
```

Comment:
looks like we now can use `$data = new Data\Collection($response);`

because it supports xml response.